src/app/client/projects/desktop/src/app/modules/offline/components/desktop-app-update/desktop-app-update.component.ts

```diff
@@ -0,0 +1,57 @@
+import { IInteractEventEdata, IInteractEventObject } from '@sunbird/telemetry';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { AppUpdateService } from './../../services/app-update/app-update.service';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import * as _ from 'lodash-es';
+
+@Component({
+  selector: 'app-desktop-app-update',
+  templateUrl: './desktop-app-update.component.html',
+  styleUrls: ['./desktop-app-update.component.scss']
+})
+export class DesktopAppUpdateComponent implements OnInit, OnDestroy {
+
+  isUpdated;
+  downloadUrl: string;
+  public unsubscribe$ = new Subject<void>();
+  telemetryInteractEdata: IInteractEventEdata;
+  public telemetryInteractObject: IInteractEventObject;
+  constructor(public appUpdateService: AppUpdateService) { }
+
+  ngOnInit() {
+    this.isAppUpdated();
+    this.setTelemetry();
+  }
+
+  isAppUpdated() {
+    this.appUpdateService.isAppUpdated().pipe(takeUntil(this.unsubscribe$)).subscribe(response => {
+      this.isUpdated = _.get(response, 'result.updateAvailable');
+      if (this.isUpdated) {
```

Comment:
if condition is not required and check for update flag and downloadUrl  before showing app update